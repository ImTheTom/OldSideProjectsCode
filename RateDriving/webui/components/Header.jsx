import React, { Component } from 'react';
import Link from 'next/link';
import PropTypes from 'prop-types';

class Header extends Component {

	static get propTypes() { 
		return { 
			isLoggedIn: PropTypes.bool
		}; 
	}

	render() {
		let option;
		let option2;
		let option3;
			option = <Link href="/signin"><a>Sign In</a></Link>;
			option2 = <a href='/signup'>Sign Up</a>;
			option3 = <a href='/logout'>Log Out</a>;
		return (
			<div>
				<div className="left">
					<nav>
						<Link href="/">
							<a>Home</a>
						</Link>
						<Link href="/about">
							<a>About</a>
						</Link>
					</nav>
				</div>
				<div className="right">
					<nav>
						{option}
						{option2}
						{option3}
					</nav>
				</div>
			</div>
		);
	}
}

export default Header;
