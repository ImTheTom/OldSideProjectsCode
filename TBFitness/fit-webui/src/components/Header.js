import React from "react";
import { Link } from "react-router-dom";

const Header = () => (
	<div className="header">
		<div className="header-links">
			<Link to="/">Home</Link>
			<Link to="/lifts">Lifts</Link>
			<Link to="/createlift">Create Lift</Link>
			<Link to="/createworkout">Create Workout</Link>
			<Link to="/workouts">Workouts</Link>
		</div>
	</div>
);

export default Header;