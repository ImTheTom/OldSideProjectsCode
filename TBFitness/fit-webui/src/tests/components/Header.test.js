import Header from "../../components/Header";
import React from "react";
import { shallow } from "enzyme";

test("should render create Header Component correctly", () => {
	const wrapper = shallow(<Header />);
	expect(wrapper).toMatchSnapshot();
});
