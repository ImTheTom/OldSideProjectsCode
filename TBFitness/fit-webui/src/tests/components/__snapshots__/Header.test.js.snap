// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render create Header Component correctly 1`] = `
<div>
  <div>
    <Link
      replace={false}
      to="/"
    >
      Home
    </Link>
    <Link
      replace={false}
      to="/lifts"
    >
      Lifts
    </Link>
    <Link
      replace={false}
      to="/createlift"
    >
      Create Lift
    </Link>
    <Link
      replace={false}
      to="/workouts"
    >
      Workouts
    </Link>
  </div>
  <div>
    Fitness.TomBowyer
  </div>
</div>
`;
